log-window.js

```diff
@@ -26,261 +26,87 @@ class EmeLogWindow {
   }
 
   /** Open the log window. */
-  open() {
+  async open() {
     if (!this.isOpen()) {
-      this.logWindow_ = window.open(
-          'log.html', 'EME Log', 'width=700,height=600');
+      this.logWindow_ = await chrome.windows.create({
+        url: chrome.runtime.getURL('log.html'),
+        type: 'popup',
+        height: 600,
+        width: 700,
+      });
 
       // Inject a copy of this class into the window so that it can get the log
       // URI later.
       this.logWindow_.EmeLogWindow = EmeLogWindow;
     }
 
     // Bring the window to the foreground.
-    this.logWindow_.focus();
+    chrome.windows.update(this.logWindow_.id, { focused: true });
+  }
+
+  /**
+   * Close the log window.
+   * @param {Number} Window ID
+  */
+  close(windowId) {
+    if (this.logWindow_ != null && this.logWindow_.id == windowId) {
+      this.logWindow_ = null;
+      this.clear();
+    }
   }
 
   /** @return {boolean} True if the log window is open. */
   isOpen() {
-    return this.logWindow_ != null && !this.logWindow_.closed;
+    return this.logWindow_ != null;
   }
 
-  /** @return {string} A URI to download the log as a text file. */
-  getTextLogUri() {
-    const blob = new Blob([this.textLogs_], {type: 'text/plain'});
-    return URL.createObjectURL(blob);
+  /** @return {string} The text log. */
+  getTextLogs() {
+    return this.textLogs_;
   }
 
   /** Clear the log window. */
   clear() {
-    const document = this.logWindow_.document;
-
-    const list = document.getElementById('eme-log');
-    while (list.hasChildNodes()) {
-      list.removeChild(list.firstChild);
-    }
-
     this.textLogs_ = '';
   }
 
-  /**
-   * @param {Object} The serialized log to format in HTML.
-   */
-  appendLog(log) {
-    if (!this.logWindow_) {
-      return;
-    }
-
-    const document = this.logWindow_.document;
-    const logElement = document.querySelector('#eme-log');
-    const li = document.createElement('li');
-    logElement.appendChild(li);
-
-    const heading = document.createElement('h3');
-    li.appendChild(heading);
-
-    const time = document.createElement('div');
-    time.classList.add('time');
-    heading.appendChild(time);
-    heading.appendChild(document.createElement('br'));
-
-    const instanceId = document.createElement('div');
-    instanceId.classList.add('instance-id');
-    heading.appendChild(instanceId);
-    heading.appendChild(document.createElement('br'));
-
-    const title = document.createElement('div');
-    title.classList.add('title');
-    heading.appendChild(title);
-
-    const timestamp = new Date(log.timestamp);
-    const formattedTimestamp = timestamp.toString();
-
-    time.textContent = formattedTimestamp;
-    if (log.duration) {
-      time.textContent += ` - duration: ${log.duration.toFixed(1)} ms`;
+  /** @param {Object} The text log. */
```

Comment:
I think this is a string, not an Object.

Reply:
Done